apps/explorer/components/TroubleshooterForm/index.tsx

```diff
@@ -0,0 +1,156 @@
+'use client'
+
+import { useCallback, useState } from 'react'
+import { useRouter } from 'next/navigation'
+import { useForm } from 'react-hook-form'
+
+import {
+  Button,
+  ConfigFields,
+  ControlGroup,
+  FieldText,
+  Heading,
+  LoadingDots,
+  Panel,
+  Text,
+  triggerErrorToast,
+} from '@siafoundation/design-system'
+import { to } from '@siafoundation/request'
+
+import { network } from '../../config/index'
+import { routes } from '../../config/routes'
+import { useExplored } from '../../hooks/useExplored'
+
+import { ContentLayout } from '../ContentLayout'
+
+const defaultValues = {
+  query: '',
+}
+
+const fields: ConfigFields<typeof defaultValues, never> = {
+  query: {
+    type: 'text',
+    title: 'Host Pubkey',
+    placeholder: 'ed25519:707d18...',
+    validation: {
+      required: 'Please input a valid host public key',
+      minLength: {
+        value: 72,
+        message: 'Please input a valid host public key (72 characters long)',
+      },
+    },
+  },
+}
+
+export function TroubleshooterForm() {
+  const [isFetching, setIsFetching] = useState(false)
+  const router = useRouter()
+  const explored = useExplored()
+
+  const form = useForm({
+    defaultValues: defaultValues,
+  })
+
+  const onSubmit = useCallback(
+    async (values: { query: string }) => {
+      setIsFetching(true)
```

Comment:
The form has this built in, its called `isSubmitting` https://react-hook-form.com/docs/useform/formstate